### PR TITLE
[SYCL] Fix isordered built-in function realisation

### PR DESCRIPTION
**Pull request by maximdimakov**

### CODE REVIEW

File: sycl/source/detail/builtins_integer.cpp [hunk 1]
```diff
@@ -171,7 +171,7 @@ template <typename T> inline T __s_long_mad_sat(T a, T b, T c) {
   if (!neg_prod && mulhi != 0)
     return d::max_v<T>();
   if (neg_prod && mulhi != -1)
-    return d::max_v<T>(); // essentially some other negative value.
+    return d::min_v<T>(); // essentially some other negative value.
```
Comment (s-kanaev) [inline sycl/source/detail/builtins_integer.cpp:174]: This change is already comitted with intel/llvm#2790 and is irrelevant to the patch.
Could you, please, remove it from the patch?

File: sycl/test/regression/mad_sat.cpp [hunk 2]
```diff
@@ -0,0 +1,138 @@
+// RUN: %clangxx -fsycl %s -o %t.out
```
Comment (s-kanaev) [inline sycl/test/regression/mad_sat.cpp:1]: This change is already comitted with intel/llvm#2790 and is irrelevant to the patch.
Could you, please, remove it from the patch?

**Review by s-kanaev — APPROVED** [review]
The test looks good.
Though, I believe, it should be augmented with checks for half and double types.

**Review by againull — APPROVED** [review]

**Pull request merged**